### ISSUE TITLE
optionalLeftJoin with executeSelectPage error

### ISSUE THREAD

**Issue by BOHDANff** [juanluispaz/ts-sql-query#54]
Hello, hope you are doing well.

Using `optionalLeftJoin` with `executeSelectPage` cause error: 
```
 Cannot read properties of undefined (reading 'has')
[...works/common]         at SelectQueryBuilder.__executeSelectCount (/node_modules/ts-sql-query/queryBuilders/SelectQueryBuilder.js:179:52)
[...works/common]         at getCountFn (/node_modules/ts-sql-query/queryBuilders/SelectQueryBuilder.js:201:23)
[...works/common]         at /node_modules/ts-sql-query/queryRunners/ManagedTransactionPoolQueryRunner.js:28:20
[...works/common]         at processTicksAndRejections (node:internal/process/task_queues:96:5)
[...works/common]     Caused By: TypeError: Cannot read properties of undefined (reading 'has')
[...works/common]         at MySqlSqlBuilder._buildSelectWithColumnsInfo (/node_modules/ts-sql-query/sqlBuilders/AbstractSqlBuilder.js:699:44)
[...works/common]         at MySqlSqlBuilder._buildSelect (/node_modules/ts-sql-query/sqlBuilders/AbstractSqlBuilder.js:551:29)
[...works/common]         at SelectQueryBuilder.__buildSelectCount (/node_modules/ts-sql-query/queryBuilders/SelectQueryBuilder.js:636:34)
[...works/common]         at SelectQueryBuilder.__executeSelectCount (/node_modules/ts-sql-query/queryBuilders/SelectQueryBuilder.js:171:32)
[...works/common]         at getCountFn (/node_modules/ts-sql-query/queryBuilders/SelectQueryBuilder.js:201:23)
[...works/common]         at /node_modules/ts-sql-query/queryRunners/ManagedTransactionPoolQueryRunner.js:28:20
[...works/common]         at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

How I understand the problem is in `/ts-sql-query/src/sqlBuilders/AbstractSqlBuilder.ts`  712 str
```js
if (!requiredTablesOrViews.has(join.__tableOrView))
```
where `requiredTablesOrViews` comes as `undefined`.

Also I found that this problems is caused by method which make count request, because `executeSelectMany` doesn't cause such error.

Small example which represents this problem:
```js
class DBConection extends MySqlConnection<'DBConnection'> { }

const tProduct = new class TProduct extends Table<DBConection, 'TProduct'> {
    id = this.autogeneratedPrimaryKey('id', 'int')
    name = this.column('name', 'string')
  }()

const tProductGroup = new class TProduct extends Table<DBConection, 'TProductGroup'> {
    id = this.autogeneratedPrimaryKey('id', 'int')
    name = this.column('name', 'string')
    constructor() {
        super('product_group'); // table name in the database
    }
}()

const productGroup = tProductGroup.forUseInLeftJoin();

const availableFields = {
     id: tProduct.id,
     productName: tProduct.name,
     groupName: productGroup.name,
};

const fieldsToPick = ['productName', 'groupName'];

const pickedFields = dynamicPick(availableFields, fieldsToPick, ['id']);

const productSelect = connection.selectFrom(tProduct)
        .optionalLeftJoin(productGroup).on(tProduct.groupId.equals(productGroup.id))
        .select(pickedFields)
        })
        .executeSelectPage()

```

**Comment by juanluispaz** [comment]
Fixed in ts-sql-query 1.41.0 already published. Sorry for the delay

**Comment by BOHDANff** [comment]
Thank you very much!
